File: backend/amoz/src/main/java/com/zpi/amoz/requests/StockCreateRequest.java
package com.zpi.amoz.requests;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;

import java.util.Optional;

@Schema(description = "Request do tworzenia informacji o stanie magazynowym produktu, zawierający dane dotyczące dostępnych ilości oraz ewentualnej ilości alarmowej.")
public record StockCreateRequest(

        @Schema(description = "Ilość dostępna w magazynie", example = "100")
        @NotNull(message = "Amount available is required")
        @Min(value = 0, message = "Amount available must be greater than or equal to 0")
        Integer amountAvailable,

        @Schema(description = "Ilość alarmowa, poniżej której stan magazynowy wymaga uzupełnienia", example = "10", nullable = true)
        Optional<@Min(value = 0, message = "Alarming amount must be greater than or equal to 0") Integer> alarmingAmount
) {
}

File: frontend/app/src/main/java/com/example/amoz/requests/StockCreateRequest.kt
package com.example.amoz.requests

import jakarta.validation.constraints.Min
import kotlinx.serialization.Serializable

@Serializable
data class StockCreateRequest(

    @field:Min(value = 0, message = "Amount available must be greater than or equal to 0")
    val amountAvailable: Int,

    @field:Min(value = 0, message = "Alarming amount must be greater than or equal to 0")
    val alarmingAmount: Int? = null
)
